api/bootloader/util.go

```diff
@@ -35,3 +38,28 @@ func HashFirmware(firmwareVersion uint32, unsignedFirmware []byte) []byte {
 	copy(padded, unsignedFirmware)
 	return doubleHash(append(firmwareVersionLE, padded...))
 }
+
+// ParseSignedFirmware parses a signed firmware file and returns (sigdata, firmware). Errors if the
+// format is invalid, or the firmware magic does not match the expected magic according to the
+// device product.
```

Comment:
nit in the comment: the device product magic is not checked here, but inside `device.ParseSignedFirmware`. Here it just checks that the magic is a valid one